shell/platform/common/client_wrapper/include/flutter/texture_registrar.h

```diff
@@ -95,9 +95,16 @@ class TextureRegistrar {
   // the callback that was provided upon creating the texture.
   virtual bool MarkTextureFrameAvailable(int64_t texture_id) = 0;
 
-  // Unregisters an existing Texture object.
-  // Textures must not be unregistered while they're in use.
-  virtual bool UnregisterTexture(int64_t texture_id) = 0;
+  // Asynchronously unregisters an existing texture object.
+  // Upon completion, the optional |callback| gets invoked.
+  virtual void UnregisterTexture(int64_t texture_id,
+                                 std::function<void()> callback) = 0;
+
+  // Unregisters an existing texture object.
+  [[deprecated(
+      "Use UnregisterTexture(texture_id, optional_callback) "
+      "instead.")]] virtual bool
```

Comment:
For now, just use a comment like:
https://github.com/flutter/engine/blob/6e1c88fa7676566cb051c79ef082d5f4f9ce05ae/shell/platform/windows/public/flutter_windows.h#L160

I recently landed https://github.com/flutter/engine/blob/main/shell/platform/common/public/flutter_macros.h but haven't yet updated our recipes to bundle it with the client wrapper code. Once that happens, we can use that instead.